lib/fluent/plugin/in_tail.rb

```diff
@@ -1063,12 +1234,19 @@ def handle_notify
                 begin
                   while true
                     @start_reading_time ||= Fluent::Clock.now
+                    group_watcher.update_reading_time(@path) unless group_watcher.nil?
                     data = io.readpartial(BYTES_TO_READ, @iobuf)
                     @eof = false
                     @number_bytes_read += data.bytesize
                     @fifo << data
+                    group_watcher.update_lines_read(@path, -@lines.size) unless group_watcher.nil?
                     @fifo.read_lines(@lines)
+                    group_watcher.update_lines_read(@path, @lines.size) unless group_watcher.nil?
 
+                    if !group_watcher.nil? && group_watcher.limit_lines_reached?(@path) || should_shutdown_now?
```

Comment:
Can we extract this condition as a method for simplicity and readability?

Reply:
Thanks! I have changed it to:
```ruby
if group_watcher&.limit_lines_reached?(@path) || should_shutdown_now?
```